vra7/resource_vra7_deployment.go

```diff
@@ -435,6 +445,10 @@ func resourceVra7DeploymentRead(d *schema.ResourceData, meta interface{}) error
 			resourceConfigStruct.ParentResourceID = parentResourceID
 			resourceConfigStruct.IPAddress = data["ip_address"].(string)
 
+			if p != nil && p.ResourceConfiguration != nil {
```

Comment:
Note: would the `p !- nil` check be redundant from the above error check from `readProviderConfiguration()`.